subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java

```diff
@@ -85,18 +84,6 @@ public String create() {
                             }
                         }
                     });
-                    DeprecationLogger.whileDisabled(new Runnable() {
-                        @Override
-                        public void run() {
-                            container.whenObjectRemoved(new Action<I>() {
-                                @Override
-                                public void execute(I item) {
-                                    String name = namer.determineName(item);
-                                    containerNode.removeLink(name);
```

Comment:
Is whenObjectRemoved still called when a task is replaced?

Reply:
Why yes, yes it is...fixed.